docs: Add a disclaimer to avoid using internal metrics (#1880)
kyma-project/telemetry-manager: docs/user/04-metrics.md

The Telemetry module ensures that the OTel Collector instances are operational and healthy at any time, for example, with buffering and retries. However, there may be situations when the instances drop metrics, or cannot handle the metric load.

> To detect and fix such situations, check the [pipeline status](./resources/05-metricpipeline.md#metricpipeline-status) and check out [Troubleshooting](#troubleshooting). The pipeline status is exposed via dedicated [metrics](./04-metrics.md#5-monitor-pipeline-health) of a MetricPipeline which can be alerted and reported on in an integrated backend like [SAP CloudLogging](./integration/sap-cloud-logging/README.md#use-sap-cloud-logging-alerts).

Reviewer: ```suggestion
To detect and fix such situations, check the [pipeline status](./resources/05-metricpipeline.md#metricpipeline-status) and check out [Troubleshooting](#troubleshooting). Dedicated [metrics](./04-metrics.md#5-monitor-pipeline-health) expose the pipeline status of a MetricPipeline. You can set up alerts and reports for these metrics in an integrated backend like [SAP Cloud Logging](./integration/sap-cloud-logging/README.md#use-sap-cloud-logging-alerts).
```

Reviewer: ```suggestion
To detect and fix such situations, check the [pipeline status](./resources/05-metricpipeline.md#metricpipeline-status) and check out [Troubleshooting](#troubleshooting).  If you have set up [pipeline health monitoring](./04-metrics.md#5-monitor-pipeline-health), check the alerts and reports in an integrated backend like [SAP Cloud Logging](./integration/sap-cloud-logging/README.md#use-sap-cloud-logging-alerts).
```

> > [!NOTE]

Reviewer: ```suggestion
> [! WARNING]
```

> > Accessing the metrics endpoint of the used OTel Collector instances directly is not recommended, as the exposed metrics are not treated as official API of the Kyma Telemetry module. Breaking changes can happen if the underlying OTel-Collector version introduces such. Please use the above mentioned pipeline status instead.

Reviewer: ```suggestion
> Don't access the metrics endpoint of the used FluentBit instances directly, because the exposed metrics are no official API of the Kyma Telemetry module. Breaking changes can happen if the underlying FluentBit version introduces such.
> Instead, use the [pipeline status](./resources/02-logpipeline.md#logpipeline-status).
```

The Telemetry module ensures that the OTel Collector instances are operational and healthy at any time, for example, with buffering and retries. However, there may be situations when the instances drop metrics, or cannot handle the metric load.

To detect and fix such situations, check the [pipeline status](./resources/05-metricpipeline.md#metricpipeline-status) and check out [Troubleshooting](#troubleshooting). The pipeline status is exposed via dedicated [metrics](./04-metrics.md#5-monitor-pipeline-health) of a MetricPipeline which can be alerted and reported on in an integrated backend like [SAP CloudLogging](./integration/sap-cloud-logging/README.md#use-sap-cloud-logging-alerts).

> [! WARNING]

> > Don't access the metrics endpoint of the used OTel Collector instances directly, because the exposed metrics are no official API of the Kyma Telemetry module. Breaking changes can happen if the underlying OTel Collector version introduces such.

Reviewer: ```suggestion
> It's not recommended to access the metrics endpoint of the used OTel Collector instances directly, because the exposed metrics are no official API of the Kyma Telemetry module. Breaking changes can happen if the underlying OTel Collector version introduces such.
```
